sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java

```diff
@@ -701,6 +705,90 @@ void delete(String containerName,
+                    Map<String, String> tags = null;
+                    if (result.getBlobTagSet() != null) {
+                        tags = new HashMap<>();
+                        for (BlobTag tag : result.getBlobTagSet()) {
+                            tags.put(tag.getKey(), tag.getValue());
+                        }
+                    }
+                    callback.onSuccess(tags, header, response);
+                }
+
+                @Override
+                public void onFailure(Throwable throwable, Response response) {
+                    callback.onFailure(throwable, response);
+                }
+            });
+    }
+
+    /**
+     * Gets tags associated with a blob.
+     *
+     * @param containerName     The container name.
+     * @param blobName          The blob name.
+     * @param snapshot          The snapshot parameter is an opaque DateTime value that, when present, specifies the
+     *                          blob snapshot to retrieve. For more information on working with blob snapshots, see &lt;a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/creating-a-snapshot-of-a-blob"&gt;Creating a Snapshot of a Blob.&lt;/a&gt;.
+     * @param timeout           The timeout parameter is expressed in seconds. For more information, see
+     *                          &lt;a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations"&gt;Setting Timeouts for Blob Service Operations.&lt;/a&gt;.
+     * @param version           Specifies the version of the operation to use for this request.
+     * @param requestId         Provides a client-generated, opaque value with a 1 KB character limit that is
+     *                          recorded in the analytics logs when storage analytics logging is enabled.
+     * @param cancellationToken The token to request cancellation.
+     * @param callback          The callback that receives the response.
+     */
+    public void getBlobTags(String containerName,
+                            String blobName,
+                            String snapshot,
+                            Integer timeout,
+                            String version,
+                            String requestId,
+                            CancellationToken cancellationToken,
+                            CallbackWithHeader<Map<String, String>, BlobGetTagsHeaders> callback) {
+        this.storageBlobServiceClient.getTags(containerName,
+            blobName,
+            snapshot,
+            null,
+            timeout,
+            version,
+            null,
+            requestId,
+            cancellationToken,
+            new CallbackWithHeader<BlobTags, BlobGetTagsHeaders>() {
+                @Override
+                public void onSuccess(BlobTags result, BlobGetTagsHeaders header, Response response) {
+                    Map<String, String> tags = null;
```

Comment:
Should we return empty map or null here if there are no tags?

Reply:
We should probably bump the XMS_VERSION to the 73 version since gettags was not supported in 2019-02-02 I dont think

Reply:
I would think an empty map, but there might be a guideline for this

Reply:
It looks like tagset will never be null from the generated code so I think this is fine?